modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java

```diff
@@ -16,114 +16,189 @@
+     * new URLs and those ones consuming them.
+     *
+     * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+     */
+    static class NewUrlsCollector implements Action1<Resource> {
+        private List<PingDestination> newUrls = new ArrayList<>();
+
+        /**
+         * A callback for the {@link Inventory} that collects newly added URLs. It is safe to call this method
+         * concurrently from any random thread.
+         *
+         * @see rx.functions.Action1#call(java.lang.Object)
+         */
+        @Override
+        public void call(Resource r) {
+            if (PingDestination.isUrl(r)) {
+                synchronized (this) {
+                    newUrls.add(PingDestination.from(r));
+                }
+            }
+        }
+
+        /**
+         * Returns the list of {@link PingDestination}s collected by this {@link NewUrlsCollector}. It is safe to call
+         * this method concurrently from any random thread.
+         *
+         * @return the list of {@link PingDestination}s
+         */
+        public List<PingDestination> getNewUrls() {
+            synchronized (this) {
+                if (this.newUrls.isEmpty()) {
+                    return Collections.emptyList();
+                } else {
+                    List<PingDestination> result = this.newUrls;
+                    this.newUrls = new ArrayList<>();
+                    return result;
+                }
+            }
+        }
+    }
+
     /** How many rounds a WAIT_MILLIS do we wait for results to come in? */
     private static final int ROUNDS = 15;
     /** How long do we wait between each round in milliseconds */
     private static final int WAIT_MILLIS = 500;
-    /** Rough timeout in milliseconds for the pings afher which the pings are cancelled and reported as timeouted.
-     * Note that in practice, the real time given to pings can be longer. */
+    /**
+     * Rough timeout in milliseconds for the pings after which the pings are cancelled and reported as timeouted. Note
+     * that in practice, the real time given to pings can be longer.
+     */
     private static final int TIMEOUT_MILLIS = ROUNDS * WAIT_MILLIS;
 
-    private final String tenantId = "test";
-
     @EJB
-    public Pinger pinger;
+    Pinger pinger;
 
-    Set<PingDestination> destinations = new HashSet<>();
+    private final Set<PingDestination> destinations = new HashSet<>();
 
     @EJB
-    public MetricPublisher metricPublisher;
+    MetricPublisher metricPublisher;
+
+    @EJB
+    TraitsPublisher traitsPublisher;
+
+    @Inject
+    @Observable
+    private Inventory.Mixin.Observable inventory;
+
+    final NewUrlsCollector newUrlsCollector = new NewUrlsCollector();
 
-    private final PingerConfiguration configuration = PingerConfiguration.defaults();
+    private void addUrl(String tenantId, String envId, String url) {
+        Log.LOG.infof("Putting url to inventory: %s", url);
+
+        Single tenant = null;
```

Comment:
This whole setup block for tenant, env, and resource type should be unnecessary, because these are set up automagically by glue (hawkular-integrated-inventory-rest/TemporaryHacks).